lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -729,6 +730,172 @@ LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
+  auto reg =
+      createRegister(loc, rewriter, getComponent(), 1,
+                     getState<ComponentLoweringState>().getUniqueName(opName));
+
+  // Operation pipelines are not combinational, so a GroupOp is required.
+  auto group = createGroupForOp<calyx::GroupOp>(rewriter, cmpf);
+  OpBuilder builder(group->getRegion(0));
+  getState<ComponentLoweringState>().addBlockScheduleable(cmpf->getBlock(),
+                                                          group);
+
+  rewriter.setInsertionPointToEnd(group.getBodyBlock());
+  rewriter.create<calyx::AssignOp>(loc, calyxCmpFOp.getLeft(), cmpf.getLhs());
+  rewriter.create<calyx::AssignOp>(loc, calyxCmpFOp.getRight(), cmpf.getRhs());
+
+  // Prepare signals and create registers
+  SmallVector<calyx::RegisterOp> inputRegs;
+  for (const auto &input : info.inputPorts) {
+    Value signal;
+    switch (input.port) {
+    case Port::eq: {
+      signal = calyxCmpFOp.getEq();
+      break;
+    }
+    case Port::gt: {
+      signal = calyxCmpFOp.getGt();
+      break;
+    }
+    case Port::lt: {
+      signal = calyxCmpFOp.getLt();
+      break;
+    }
+    case Port::unordered: {
+      signal = calyxCmpFOp.getUnordered();
+      break;
+    }
+    }
+    std::string nameSuffix =
+        (input.port == PredicateInfo::InputPorts::Port::unordered)
+            ? "unordered_port"
+            : "compare_port";
+    auto signalReg =
+        createSignalRegister(signal, input.invert, nameSuffix, loc, rewriter,
+                             getComponent(), calyxCmpFOp, group, builder);
+    inputRegs.push_back(signalReg);
+  }
+
+  // Create the output logical operation
+  Value outputValue, doneValue;
+  switch (info.logic) {
+  case CombLogic::noComb: {
+    // it's guaranteed to be either ORD or UNO
+    outputValue = inputRegs[0].getOut();
+    doneValue = inputRegs[0].getOut();
+    break;
+  }
+  case CombLogic::logicAnd: {
+    auto outputLibOp = getState<ComponentLoweringState>()
+                           .getNewLibraryOpInstance<calyx::AndLibOp>(
+                               rewriter, loc, {one, one, one});
+    rewriter.create<calyx::AssignOp>(loc, outputLibOp.getLeft(),
+                                     inputRegs[0].getOut());
+    rewriter.create<calyx::AssignOp>(loc, outputLibOp.getRight(),
+                                     inputRegs[1].getOut());
+
+    outputValue = outputLibOp.getOut();
+    break;
+  }
+  case CombLogic::logicOr: {
+    auto outputLibOp = getState<ComponentLoweringState>()
+                           .getNewLibraryOpInstance<calyx::OrLibOp>(
+                               rewriter, loc, {one, one, one});
+    rewriter.create<calyx::AssignOp>(loc, outputLibOp.getLeft(),
+                                     inputRegs[0].getOut());
+    rewriter.create<calyx::AssignOp>(loc, outputLibOp.getRight(),
+                                     inputRegs[1].getOut());
+
+    outputValue = outputLibOp.getOut();
```

Comment:
should there be a `break` here in case a default / another case is added?

Reply:
good point